DataStructures/Graphs/BFS.java

```diff
@@ -0,0 +1,47 @@
+import java.util.*;
+
+public class BFSs {
+
+	public static void main(String[] args) {
+		Scanner s = new Scanner(System.in);
+		int V = s.nextInt();
+		int E = s.nextInt();
+
+		int edges[][]=new int[V][V];
+        for(int i=0;i<E;i++){
+            int fv=s.nextInt();
+            int sv=s.nextInt();
+            edges[fv][sv]=1;
+            edges[sv][fv]=1;
+        }
+        printBFS(edges);
+	}
+    
+    private static void printBFS(int edges[][]){
+        boolean visited[] = new boolean[edges.length];
+        // printBFS(edges,0,visited);
+		for(int i=0;i<edges.length;i++) {
+			if(!visited[i]) {
+				printBFS(edges,i,visited);
+			}
+		}
+    }
+    
+    private static void printBFS(int edges[][],int sv,boolean visited[]){
+        Queue<Integer> q=new LinkedList<>();
+        q.add(sv);
+        int n=edges.length;
+        visited[sv]=true;
+        String s=new String();
```

Comment:
Our static analysis engine detects that your pull request includes a change that **method invokes inefficient new String() constructor**. According to [the best practice](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-inefficient-new-string-constructor-dm-string-void-ctor), creating a new java.lang.String object using the no-argument constructor **wastes memory** because the object so created will be functionally indistinguishable from the empty string constant "".  Java guarantees that identical string constants will be represented by the same String object.  **Therefore, you should just use the empty string constant directly.**